api/main_endpoints/routes/Event.js

```diff
@@ -12,14 +12,35 @@ const {
   FORBIDDEN,
   NOT_FOUND
 } = require('../../util/constants').STATUS_CODES;
-const addErrorLog = require ('../util/logging-helpers');
+const addErrorLog = require('../util/logging-helpers');
 const membershipState = require('../../util/constants').MEMBERSHIP_STATE;
 
 router.get('/getEvents', (req, res) => {
```

Comment:
the api handlers `/getEvents` and `/getUpcomingEvents` effectively do the same thing with a minor difference.

to save code we can abstract it to a function

something like:
```js
function eventQueryHandler(res, date=false) {
    let maybeQuery = {}
  if (date) {
    // insert logic here for calculating today
    maybeQuery = {
      eventDate: {
        $gte: today.toISOString(), // gets all upcoming events including today's
      },
    };
  }
  Event.find(maybeQuery)
      // paste the rest of the query here, there the same in both handler

}
```

`date` specifies if we want events today and beyond, res is the response to the user. the two functions become:

```js
router.get('/getEvents', (req, res) => {
  getAllEvents(res);
});

router.get('/getUpcomingEvents', (req, res) => {
  getAllEvents(res, true);
});
```

Reply:
with this change the behavior should stay exactly the same and we cut down on code

Reply:
I simplified the two get event requests like this: https://github.com/SCE-Development/Core-v4/pull/933/commits/b841b5a537ca1baae2402ffee63ba366e9ed9279
> fixed syntax issues in the next 2 commits

```js
function getEvents(req, res, upcomingEvents) {
  let query = {};
  if(upcomingEvents) {
    const today = new Date(Date.now());
    today.setHours(0, 0, 0, 0);
    query = {
      eventDate: { $gte: today.toISOString() }
    };
  }
  Event.find(query)
    .sort({ eventDate: -1, startTime: -1 })
    .then((items) => res.status(OK).send(items))
    .catch((error) => {
      const info = {
        errorTime: new Date(),
        apiEndpoint: 'Event/getEvents',
        errorDescription: error,
      };
      addErrorLog(info);
      res.status(BAD_REQUEST).send({ error, message: 'Getting event failed' });
    });
}

router.get('/getEvents', (req, res) => {
  getEvents(req, res, false);
});

router.get('/getUpcomingEvents', (req, res) => {
  getEvents(req, res, true);
});
```